cpp/include/cudf/lists/detail/scatter.cuh

```diff
@@ -223,7 +223,7 @@ std::unique_ptr<column> scatter(
   auto const num_rows = target.size();
   if (num_rows == 0) { return cudf::empty_like(target); }
 
-  auto lv        = static_cast<list_scalar const*>(&slr);
+  auto lv        = dynamic_cast<list_scalar const*>(&slr);
```

Comment:
C.148: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-ptr-cast

> A failure to find the required class will cause `dynamic_cast` to return a null value, and de-referencing a null-valued pointer will lead to undefined behavior. Therefore the result of the `dynamic_cast` should always be treated as if it might contain a null value, and tested.

Should we instead `dynamic_cast` to a reference type? See C.147: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c147-use-dynamic_cast-to-a-reference-type-when-failure-to-find-the-required-class-is-considered-an-error

```suggestion
  auto lv        = dynamic_cast<list_scalar const&>(slr);
```
(and then change semantics below to reference `.` instead of pointer `->`)